### PR TITLE
fix: Should rotate agent log

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

DayRotator

##### Description of change

because schedule is running under app worker,
agent log is the only differece between app worker and agent worker.
- app worker -> egg-web.log
- agent worker -> egg-agent.log

### CODE REVIEW

File: test/fixtures/logrotator-agent/app.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+
+module.exports = function(app) {
+  // won't find egg-agent from rotateLogDirs
+  app.config.logger.rotateLogDirs = [];
```
Comment (popomore) [inline test/fixtures/logrotator-agent/app.js]: 这个就是我今天说的问题，如果去掉了 rotateLogDirs，schedule 是不会切 egg-agent 的

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
2.2.2

File: app/lib/day_rotator.js [hunk 2]
```diff
@@ -22,6 +22,13 @@ class DayRotator extends Rotator {
     for (const key in loggers) {
       this._setFile(loggers[key].options.file, files);
     }
+    // Should rotate agent log, because schedule is running under app worker,
+    // agent log is the only differece between app worker and agent worker.
+    // - app worker -> egg-web.log
+    // - agent worker -> egg-agent.log
+    const logDir = this.app.config.logger.dir;
+    const agentLogName = this.app.config.logger.agentLogName;
```
Comment (atian25) [inline app/lib/day_rotator.js]: 这个 agentLogName 的默认值是在哪里?

Comment (atian25) [inline app/lib/day_rotator.js]: 找到了 https://github.com/eggjs/egg/blob/master/config/config.default.js#L158

**Pull request merged**